src/components/text-editor/utils/markdown-converter.ts

```diff
@@ -1,14 +1,25 @@
 import { ContentTypeConverter } from './content-type-converter';
 import { EditorView } from 'prosemirror-view';
+import { Node as ProseMirrorNode } from 'prosemirror-model';
 import {
     MarkdownSerializer,
+    MarkdownSerializerState,
     defaultMarkdownSerializer,
 } from 'prosemirror-markdown';
 import { markdownToHTML } from '../../markdown/markdown-parser';
 
+const mentionSerializer = {
+    mention: (state: MarkdownSerializerState, node: ProseMirrorNode) => {
+        state.write(
+            `<limebb-mention type="${node.attrs.type}" objectid="${node.attrs.objectid}" descriptive="${node.attrs.descriptive}"></limebb-mention>`,
```

Comment:
I don't think we should hardcode this element here like this. For this to be reusable, we would either have to provide some config or a factory to the the text editor component that will create the element that the consumer needs

Reply:
Absolutely